pkg/provision/primitives/converter.go

```diff
@@ -208,6 +208,25 @@ func K8SToProvisionType(w workloads.Workloader) (Kubernetes, string, error) {
 	return k8s, k.NodeId, nil
 }
 
+// VMToProvisionType converts type to internal provision type
+func VMToProvisionType(w workloads.Workloader) (VM, string, error) {
+	k, ok := w.(*workloads.VirtualMachine)
+	if !ok {
+		return VM{}, "", fmt.Errorf("failed to convert kubernetes workload, wrong format")
+	}
+
+	vm := VM{
```

Comment:
Since this is so similar to K8s strcut i suggest that you change Kubernetes struct to be like
```go
type Kubernetes struct{
   VM
   ...
    // extra k8s specific properties
}
```